pkg/managed/deployment_test.go

```diff
@@ -0,0 +1,171 @@
+	assert := assert.New(t)
+
+	modelBindingPair := testutil.GetModelBindingPair()
+	clusterConnections := testutil.GetManagedClusterConnections()
+	clusterConnection := clusterConnections["cluster1"]
+	manifest := testutil.GetManifest()
+	vmiSecret := monitoring.NewVmiSecret(modelBindingPair.Binding)
+	secrets := &testutil.FakeSecrets{Secrets: map[string]*corev1.Secret{
+		constants.VmiSecretName: vmiSecret,
+	}}
+
+	// set env vars needed for test; set them back when the test completes
+	defer setEnv(t, "WLS_MICRO_REQUEST_MEMORY", os.Getenv("WLS_MICRO_REQUEST_MEMORY"))
+	setEnv(t, "WLS_MICRO_REQUEST_MEMORY", "2.5Gi")
+	defer setEnv(t, "COH_MICRO_REQUEST_MEMORY", os.Getenv("COH_MICRO_REQUEST_MEMORY"))
+	setEnv(t, "COH_MICRO_REQUEST_MEMORY", "2.5Gi")
+	defer setEnv(t, "HELIDON_MICRO_REQUEST_MEMORY", os.Getenv("HELIDON_MICRO_REQUEST_MEMORY"))
+	setEnv(t, "HELIDON_MICRO_REQUEST_MEMORY", "2.5Gi")
+
+	err := CreateDeployments(modelBindingPair, clusterConnections, &manifest, "testURI", secrets)
+	assert.Nil(err, "got an error from CreateDeployments: %v", err)
+
+	assertDeployments(err, clusterConnection, assert)
+}
+
+// TestCreateDeployments tests that an existing deployment is properly updated.
+// GIVEN a cluster which has an existing expected deployment (verrazzano-wko-operator)
+//  WHEN I call CreateDeployments
+//  THEN there should be an expected set of deployments created
+//    AND the existing deployment (verrazzano-wko-operator) should be updated as expected
+func TestCreateDeploymentsUpdateExisting(t *testing.T) {
+	assert := assert.New(t)
+
+	modelBindingPair := testutil.GetModelBindingPair()
+	clusterConnections := testutil.GetManagedClusterConnections()
+	clusterConnection := clusterConnections["cluster1"]
+	manifest := testutil.GetManifest()
+	vmiSecret := monitoring.NewVmiSecret(modelBindingPair.Binding)
+	secrets := &testutil.FakeSecrets{Secrets: map[string]*corev1.Secret{
+		constants.VmiSecretName: vmiSecret,
+	}}
+
+	// set env vars needed for test; set them back when the test completes
+	defer setEnv(t, "WLS_MICRO_REQUEST_MEMORY", os.Getenv("WLS_MICRO_REQUEST_MEMORY"))
+	setEnv(t, "WLS_MICRO_REQUEST_MEMORY", "2.5Gi")
+	defer setEnv(t, "COH_MICRO_REQUEST_MEMORY", os.Getenv("COH_MICRO_REQUEST_MEMORY"))
+	setEnv(t, "COH_MICRO_REQUEST_MEMORY", "2.5Gi")
+	defer setEnv(t, "HELIDON_MICRO_REQUEST_MEMORY", os.Getenv("HELIDON_MICRO_REQUEST_MEMORY"))
+	setEnv(t, "HELIDON_MICRO_REQUEST_MEMORY", "2.5Gi")
+
+	deployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "verrazzano-wko-operator",
+			Namespace: "verrazzano-system",
+		},
+	}
+	_, err := clusterConnection.KubeClient.AppsV1().Deployments("verrazzano-system").Create(context.TODO(), &deployment, metav1.CreateOptions{})
+	assert.Nil(err, "got an error creating deployment: %v", err)
+
+	err = CreateDeployments(modelBindingPair, clusterConnections, &manifest, "testURI", secrets)
+	assert.Nil(err, "got an error from CreateDeployments: %v", err)
```

Comment:
This is a little bit unclear to me...likely because my knowledge of the operators is not that good. It looks like you are just deploying the verrazzano-wko-operator, but in assertDeployments it is asserting for "verrazzano-coh-cluster-operator" and "verrazzano-helidon-app-operator" in addition to the wko operator, as well as the prom pusher in the monitoring ns. 
Does deploying the wko operator result in all these being created, or am I missing something else in how it works?

Reply:
Oh, or maybe the model binding pair has a coherence binding and a helidon binding, and the verrazzano operator is expected to deploy those operators?

Reply:
Yes, `newDeployments` in `deployment.go` does that based on the managed clusters in the model binding pair...
```
	// Does a WebLogic application need to be deployed to this cluster?  If so, deploy the micro-operator that will manage it.
	if managedCluster.WlsOperator != nil {
...
	}

	// Does a Coherence applications need to be deployed to this cluster?  If so, deploy the micro-operator that will manage it.
	if managedCluster.CohOperatorCRs != nil {
...
	}

	// Does a Helidon Application need to be deployed to this cluster?  If so, deploy the micro-operator that will manage it.
	if managedCluster.HelidonApps != nil && manifest.HelidonAppOperatorImage != "" {
...
	}
```